zbsprimarystorage/zbsprimarystorage/zbsagent.py

```
@@ -104,6 +101,61 @@ def __init__(self):
+        if lp.redundanceAndPlaceMentPolicy:
+            try:
+                d = base64.b64decode(lp.redundanceAndPlaceMentPolicy)
+                self.redundanceAndPlaceMentPolicy = self.RedundanceAndPlacementPolicy(
+                    copysetNum=d.copysetNum,
+                    replicaNum=d.replicaNum,
+                    zoneNum=d.zoneNum
+                )
+            except Exception as e:
+                raise Exception('failed to base64 decode redundanceAndPlaceMentPolicy[%s], error[%s].' % (
+                    lp.redundanceAndPlaceMentPolicy, e.message
+                ))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)